aips/account_abstraction.md

```diff
@@ -0,0 +1,204 @@
+---
+aip: 
+title: Account Abstraction
+author: lightmark
+discussions-to (*optional): <a url pointing to the official discussion thread>
+Status: Draft
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: Framework
+created: 10/01/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): <AIP number(s)>
+---
+
+# AIP-X - Account Abstraction
+
+## Summary
+
+Account Abstraction (AA) on Aptos allows any account to be authenticated through move code in addition to existing 
+authentication schemes supported by native code. This AIP will introduce a new authenticator called `Abstraction` that
+delegates the authentication verification to AptosVM running move-based authentication logic.
+
+### Out of scope
+
+How to properly design an authentication function/module is out of scope.
+The `PayMaster`, aka, online gas payer, is not included in the design of AA on Aptos.
+
+## High-level Overview
+The core concept is to enable the move function to authenticate account signers, moving beyond the limitations of native Rust authenticators, which only support a restricted set of cryptographic schemes. To accomplish this, we propose transferring the authentication process from native Rust code to the AptosVM layer for each abstracted account. The solution we propose involves the following steps:
+
+1. Store the `FunctionInfo` of the customized Move authentication function in a new resource at the AA (Abstract Account) address.
+2. When the account uses an AbstractionAuthenticator, the Move function stored in step 1 will be executed (leveraging native dynamic dispatch) to perform the authentication check.
+
+### Why is this approach superior to other options?
+
+- No major security concerns: Users who opt for AA are responsible for managing their account’s security.
+- Scalable: This approach can easily scale as it decouples authentication from rigid native cryptographic checks and supports multiple authentication functions.
+- Audit-proof and safe: The dispatchable function mechanism has already been audited and is widely used in the FA standard, ensuring its security.
+- Rich transaction context: The Transaction Context already provides comprehensive information about the current transaction, which can be leveraged for robust authentication.
+
+This solution enhances flexibility and scalability while maintaining a secure and efficient authentication process.
+
+## Impact
+
+Account Abstraction(AA) aims to support dispatchable authentication **at the level of smart contract.**
+The account can set up in such a way that the transaction can be authorized with complicated move smart contract instead of native account authenticators provided by Aptos blockchain core, enabling a series of possibilities:
+
+- Multisig v2 account
+- Sandbox or temporary account to isolate assets/risks, etc.
+- Restricted permissions granted to trusted applications so users don’t need to sign every transaction.
```

Comment:
Would it be more appropriate to include a provision for resource locks here? I made a similar suggestion in the [LightAccount AIP](https://github.com/aptos-foundation/AIPs/pull/467#discussion_r1785581602), but I believe this feature could be integrated into this proposal as well.